src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-add-node.ts

```diff
@@ -241,116 +391,176 @@ class DialogZWaveJSAddNode extends LitElement {
+      await grantSecurityClasses(
+        this.hass,
+        this._entryId!,
+        this._securityClasses
+      );
+    } catch (err) {
+      this._error = err.message;
+      this._status = "grant_security_classes";
     }
+  }
 
-    if (message.event === "interview completed") {
-      this._status = "finished";
-      this._unsubscribe();
+  private _startManualInclusion() {
+    if (!this._inclusionStrategy) {
+      this._inclusionStrategy = InclusionStrategy.Default;
     }
+    this._startInclusion();
+  }
 
-    if (message.event === "interview stage completed") {
-      if (this._stages === undefined) {
-        this._stages = [message.stage];
-      } else {
-        this._stages = [...this._stages, message.stage];
-      }
+  private _startInclusion(): void {
+    if (!this.hass) {
+      return;
     }
+    this._subscribed = subscribeAddNode(
+      this.hass,
+      this._entryId!,
+      (message) => {
+        if (message.event === "inclusion started") {
+          this._status = "started";
+        }
+        if (message.event === "inclusion failed") {
+          this._unsubscribe();
+          this._status = "failed";
+        }
+        if (message.event === "inclusion stopped") {
+          // We either found a device, or it failed, either way, cancel the timeout as we are no longer searching
+          if (this._addNodeTimeoutHandle) {
+            clearTimeout(this._addNodeTimeoutHandle);
+          }
+          this._addNodeTimeoutHandle = undefined;
+        }
+
+        if (message.event === "validate dsk and enter pin") {
+          this._status = "validate_dsk_enter_pin";
+          this._dsk = message.dsk;
+        }
+
+        if (message.event === "grant security classes") {
+          if (this._inclusionStrategy === undefined) {
+            grantSecurityClasses(
+              this.hass,
+              this._entryId!,
+              message.requested_grant.securityClasses,
+              message.requested_grant.clientSideAuth
+            );
+            return;
+          }
+          this._requestedGrant = message.requested_grant;
+          this._securityClasses = message.requested_grant.securityClasses;
+          this._status = "grant_security_classes";
+        }
+
+        if (message.event === "device registered") {
+          this._device = message.device;
+        }
+        if (message.event === "node added") {
+          this._status = "interviewing";
+        }
+
+        if (message.event === "interview completed") {
+          this._unsubscribe();
+          this._status = "finished";
+        }
+
+        if (message.event === "interview stage completed") {
+          if (this._stages === undefined) {
+            this._stages = [message.stage];
+          } else {
+            this._stages = [...this._stages, message.stage];
+          }
+        }
+      },
+      this._inclusionStrategy
+    );
+    this._addNodeTimeoutHandle = window.setTimeout(() => {
+      this._unsubscribe();
```

Comment:
what happens if this hits while user is entering pin?

Reply:
Yep :-)